12_rpn_calculator/rpn_calculator.rb

```diff
@@ -0,0 +1,93 @@
+class RPNCalculator
+  def initialize
+    @stack = []
+    @total = 0
+  end
+
+  def push(number)
+    @stack << number
+  end
+
+  def value
+    @total
+  end
+
+  def plus
+    if @stack.length >= 2
+      @total += @stack.last(2).sum
+      @stack.pop(2)
+      @total
+    elsif @stack.length == 1
+      @total += @stack[0]
+      @stack.pop(1)
+      @total
+    elsif @stack.empty?
+      raise 'calculator is empty'
+    end
+  end
+
+  def minus
+    if @stack.length >= 2
+      @total += (@stack.last(2)[0] - @stack.last(2)[1])
+      @stack.pop(2)
+      @total
+    elsif @stack.length == 1
+      @total -= @stack[1]
+      @stack.pop(1)
+      @total
+    elsif @stack.empty?
+      raise 'calculator is empty'
+    end
+  end
+
+  def divide
+    if @stack.length >= 2
+      @total += (@stack.last(2)[0].to_f / @stack.last(2)[1])
+      @stack.pop(2)
+      @total
+    elsif @stack.length == 1
+      @total /= @stack[0]
+      @stack.pop(1)
+      @total
+    elsif @stack.empty?
+      raise 'calculator is empty'
+    end
+  end
+
+  def times
+    if @stack.length >= 2
+      @total += @stack.last(2).inject(:*)
+      @stack.pop(2)
+      @total
+    elsif @stack.length == 1
+      @total *= @stack[0]
+      @stack.pop(1)
+      @total
+    elsif @stack.empty?
+      raise 'calculator is empty'
+    end
+  end
+
+  def tokens(string)
+    string.split.map { |item| item =~ /[0-9]+/ ? item.to_i : item.to_sym }
+  end
+
+  def evaluate(string)
```

Comment:
This seems like a big & a bit hard to understand at first sight, can you refactor it to make it shorter & easier to read?